tools/github_update.py

```diff
@@ -96,118 +105,237 @@ def _get_repo_path(self):
-                details_url += 'console'
-        if not details_url:
-            raise Exception(
-                'Failed to determine URL for details link. ' +
-                'Provide either GITHUB_COMMIT_STATUS_URL or BUILD_URL in env.')
-        return details_url
-
-
-    def _get_auth_token(self):
+    def github_auth_token(self):
         github_token = os.environ.get('GITHUB_TOKEN_REPO_STATUS', '')
         if not github_token:
             github_token = os.environ.get('GITHUB_TOKEN', '')
         if not github_token:
             raise Exception(
                 'Failed to determine auth token to use with GitHub. ' +
                 'Provide either GITHUB_TOKEN or GITHUB_TOKEN_REPO_STATUS in env.')
-        encoded_tok = base64.encodestring(github_token.encode('utf-8'))
-        return encoded_tok.decode('utf-8').rstrip('\n')
+        return github_token
+
 
+class GithubAPI(object):
 
-    def _build_request(self, state, message, details_url = ''):
-        '''returns everything needed for the HTTP request, except the auth token'''
+    def __init__(self, repo_orgname, commit_sha, github_token):
+        self._repo_orgname = repo_orgname
+        self._commit_sha = commit_sha
+        self._github_token = github_token
+
+
+    def _build_post_update_request(self, context_label, state, message, details_url):
+        '''returns everything needed for an HTTP request to update repo status, except the auth token'''
+        payload = {
+            'context': context_label,
+            'state': state
+        }
+        if message:
+            payload['description'] = message
+        if details_url:
+            payload['target_url'] = details_url
         return {
             'method': 'POST',
-            'path': '/repos/{}/commits/{}/statuses'.format(
-                self._get_repo_path(),
-                self._get_commit_sha()),
+            'path': '/repos/{}/commits/{}/statuses'.format(self._repo_orgname, self._commit_sha),
             'headers': {
                 'User-Agent': 'github_update.py',
                 'Content-Type': 'application/json',
                 'Authorization': 'Basic HIDDENTOKEN'}, # replaced within update_query
-            'payload': {
-                'context': self._context_label,
-                'state': state,
-                'description': message,
-                'target_url': self._get_details_link_url(details_url)
-            }
+            'payload': payload
+        }
+
+
+    def _build_get_list_request(self):
+        '''returns everything needed for an HTTP request to retrieve repo statuses, except the auth token'''
+        return {
+            'method': 'GET',
+            'path': '/repos/{}/commits/{}/statuses'.format(self._repo_orgname, self._commit_sha),
+            'headers': {
+                'User-Agent': 'github_update.py',
+                'Content-Type': 'application/json',
+                'Authorization': 'Basic HIDDENTOKEN'} # replaced within update_query
```

Comment:
Being explicit that a token would have been provided in error logs, without exposing the token in the logs